### PR TITLE
Adds support for swiftsourceinfo

### PR DESCRIPTION
**Pull request by codeman9**
I took a stab at #496

### CODE REVIEW

**Comment by codeman9** [comment]
@googlebot I signed it!

File: swift/internal/compiling.bzl [hunk 1]
```diff
@@ -2175,6 +2188,11 @@ def output_groups_from_compilation_outputs(compilation_outputs):
             compilation_outputs.swiftmodule,
         ])
 
+    if compilation_outputs.swiftsourceinfo:
+        output_groups["swiftsourceinfo"] = depset([
+            compilation_outputs.swiftmodule,
```
Comment (segiddins) [inline swift/internal/compiling.bzl:2193]:
```suggestion
            compilation_outputs.swiftsourceinfo,
```

**Review by segiddins — APPROVED** [review]

File: swift/internal/compiling.bzl [hunk 2]
```diff
@@ -1706,6 +1706,9 @@ def compile(
             the compiler. If no interface file was produced (because the
             toolchain does not support them or it was not requested), this field
             will be None.
+        *   `swiftsourceinfo`: The `.swiftsourceinfo` file that was produced by
+            the compiler. If no source info file was produced (because the
+            toolchain does not support them), this field will be None.
```
Comment (brentleyjones) [inline swift/internal/compiling.bzl:1711]: nit: can this be after `swiftmodule` to stay alphabetical?

**Comment by keith** [comment]
do we know what Xcode uses this for yet?

**Comment by keith** [comment]
yea it would just be nice to see what this fixes / breaks so we know it's producing valid ones

**Comment by codeman9** [comment]
From a @kastiglione thread I found in slack: "it's for swiftc to emit diagnostics across module boundaries" and "so as I understand, it's useful to the IDE only indirectly (the IDE displaying the compiler's diagnostics)"

**Pull request closed without merging**